internal/workflow/processing.go

```diff
@@ -123,6 +125,11 @@ type TransferInfo struct {
 	DefaultPermanentLocationID *uuid.UUID
 }
 
+func (t *TransferInfo) Name() string {
+	s := strings.Split(filepath.Base(t.Key), ".")
+	return s[0]
+}
```

Comment:
Here's an alternative that seems to be faster and doesn't allocate the slice:

```go
func (t *TransferInfo) Name() string {
	base := filepath.Base(t.Key)
	if i := strings.Index(base, "."); i != -1 {
		return base[:i]
	}
	return base
}
```

https://gist.github.com/sevein/f071b4205170dd28482b085778d8ecff

```
$ go test -bench=. -benchmem .
goos: linux
goarch: amd64
pkg: foobar
cpu: Intel(R) Core(TM) i5-8400 CPU @ 2.80GHz
BenchmarkV1-6   	12694426	        85.29 ns/op	      48 B/op	       1 allocs/op
BenchmarkV2-6   	81417373	        12.79 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	foobar	2.239s
```